File: 프로그래머스/level1_K번째수.py
def solution(array, commands):
    answer = []
    for value in commands:
        temp = []
        for i in range(value[0] - 1, value[1]):
            temp.append(array[i])
        temp.sort()
        answer.append(temp[value[2] - 1])

    return answer

solution([1, 5, 2, 6, 3, 7, 4], [[2, 5, 3], [4, 4, 1], [1, 7, 3]])

File: 프로그래머스 복습(java)/src/level1_K번째수.java
import java.util.Arrays;

public class level1_K번째수 {
    static class Solution {
        public int[] solution(int[] array, int[][] commands) {
            int[] answer = new int[commands.length];

            for (int i = 0; i < commands.length; i++) {
                // better method
                // int[] temp = Arrays.copyOfRange(array, commands[i][0]-1, commands[i][1]);

                int[] temp = new int[commands[i][1] - commands[i][0] + 1];
                for (int j = 0; j < temp.length; j++) {
                    temp[j] = array[j + commands[i][0] - 1];
                }
                Arrays.sort(temp);
                answer[i] = temp[commands[i][2] - 1];
            }
            return answer;
        }
    }

    public static void main(String[] args) {
        Solution solution = new Solution();
        int[] array = {1, 5, 2, 6, 3, 7, 4};
        int[][] commands = {{2, 5, 3}, {4, 4, 1}, {1, 7, 3}};
        System.out.println(solution.solution(array, commands));
    }
}
